module/VuFindSearch/tests/unit-tests/src/VuFindTest/Backend/Solr/BackendTest.php

```diff
@@ -582,6 +583,45 @@ public function testWriteDocument()
         $client->expects($this->exactly(1))->method('setOptions')
             ->with(['timeout' => 60]);
         $connector = $this->getConnectorMock(['getUrl', 'write'], $client);
+        $connector->expects($this->once())->method('write')
+            ->with(
+                $this->equalTo($doc),
+                $this->equalTo('update'),
+                $this->isNull()
+            )
+            ->will(
+                $this->returnCallback(
+                    function () use ($connector) {
+                        // Call client factory for expectations to be met:
+                        $factory = $this->getProperty($connector, 'clientFactory');
+                        $factory('');
+                        return true;
+                    }
+                )
+            );
+        $connector->expects($this->once())->method('getUrl')
+            ->will($this->returnValue('http://localhost:8983/solr/core/biblio'));
+        $backend = new Backend($connector);
+        $this->assertEquals(
+            ['core' => 'biblio'],
+            $backend->writeDocument($doc, 60)
+        );
+    }
+
+    /**
+     * Test writeDocument
+     *
+     * @return void
+     */
+    public function testDelayedClientCreation()
+    {
+        $doc = new CommitDocument();
+        $client = $this->getMockBuilder(\Laminas\Http\Client::class)
+            ->onlyMethods(['setOptions'])
+            ->getMock();
+        $client->expects($this->exactly(0))->method('setOptions')
```

Comment:
Why use `exactly(0)` rather than `never()`? If it's not going to be called, why are we specifying `with` parameters? Or am I misunderstanding the order of precedence here?

Reply:
That was a mistake. Tests have now been rearranged and cleaned up.